### PR TITLE
ENH Adds Byte support for nll_loss (CPU)

### PR DESCRIPTION
**Pull request by thomasjpfan**
Addresses a part of https://github.com/pytorch/pytorch/issues/59765

This PR adds byte support for nll_loss on the CPU for `input.dim() == 2`.

CUDA support will be implemented when `nll_loss` migration to CUDA is completed in https://github.com/pytorch/pytorch/pull/60299 and https://github.com/pytorch/pytorch/pull/60097

### CODE REVIEW

File: test/test_nn.py [hunk 1]
```diff
@@ -15986,6 +15986,35 @@ def helper(input_size):
...
+            data_clone.requires_grad_()
+
+            prob = F.log_softmax(data_clone, dim=-1)
+            loss = nn.NLLLoss(reduction=reduction)
+            result = loss(prob, target.type(target_dtype))
+
+            if reduction == "none":
+                result.backward(gradient)
+            else:
+                result.backward(gradient[0])
```
Comment (jbschlosser) [inline test/test_nn.py:16008]: nit: no need for this logic or a `gradient` vector if you do `result.sum().backward()` instead- grads will still be verifiable

**Review by jbschlosser — APPROVED** [review]
Looks awesome! Just nitpicky style comments below
